### PR TITLE
e2e: knit: bootstrap the testsuite

### PR DESCRIPTION
**Pull request by fromanirh**
Add infra + very initial test to check knit is behaving as it should - and doesn't break.

Signed-off-by: Francesco Romani <fromani@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,6 +38,5 @@ push: image
 	@echo "pushing image"
 	$(RUNTIME) push quay.io/$(REPOOWNER)/$(IMAGENAME):$(IMAGETAG)
 
-.PHONY: test
-test:
-	go test -cover -v ./...
+test-e2e-knit: binaries
```
Comment (Tal-or) [inline Makefile:41]: Did you left this target without PHONY on purpose?

File: pkg/procs/info_test.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		t.Errorf("found unexpected entries: %v", pidInfos)
+	}
+}
+
+func TestSingleProcSingleThread(t *testing.T) {
+	dir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		t.Fatalf("creating temp dir %v", err)
+	}
+	//	defer os.RemoveAll(dir) // clean up
```
Comment (marcel-apf) [inline pkg/procs/info_test.go:39]: Do we need it?

**Review by marcel-apf — COMMENTED** [review]
Only a question

File: Makefile (hunk 1 above)
Comment (fromanirh) [reply Makefile:41]: nope, I missed it. Good catch.

File: pkg/procs/info_test.go (hunk 2 above)
Comment (fromanirh) [reply pkg/procs/info_test.go:39]: Yep, we do. Missed to update when I was deep(er) in dev mode. Good catch, thanks.

File: pkg/procs/info_test.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	defer os.RemoveAll(dir) // clean up
+
+	if err := makeFakeTree(dir, map[int]fakeEntry{
+		1: fakeEntry{
+			attrs: fakeAttrs{
+				"cmdline": "/usr/lib/systemd/systemd\x00--switched-root\x00--system\x00--deserialize",
+			},
+			tasks: map[int]fakeAttrs{
+				1: fakeAttrs{
+					"status": "Name:	systemd\nPid:	1\nCpus_allowed_list:	0-3\n",
```
Comment (yanirq) [inline pkg/procs/info_test.go:48]: how dependent are we on whitespaces/tabs here ?  can it ruin the test if mishandled ?

Comment (fromanirh) [reply pkg/procs/info_test.go:48]: while the business logic should not actually care, we need to use whatever the real file in `/proc` is using.

**Pull request merged**